pontoon/base/views.py

```diff
@@ -261,12 +261,13 @@ def translate(request, locale, slug, part=None, template='translate.html'):
         except Subpage.DoesNotExist:
             # If page not specified or doesn't exist
             page = pages[0]
-            locale_pages = pages.filter(resources__stats__locale=l)
```

Comment:
`locale_pages` is a nicer variable name. Instead of checking the length of `pages` down below, just set `page` to `None` and check `if page is not None`. Or, even better, just set `data['part']` directly in each of these places.